scripts/install-worker.sh

```diff
@@ -333,49 +335,49 @@ sudo systemctl disable kubelet
-  sudo sed -i s,SONOBUOY_E2E_REGISTRY,$SONOBUOY_E2E_REGISTRY,g /etc/eks/sonobuoy-e2e-registry-config
+if [[ -n "${SONOBUOY_E2E_REGISTRY}" ]]; then
+  sudo mv "${TEMPLATE_DIR}/sonobuoy-e2e-registry-config" /etc/eks/sonobuoy-e2e-registry-config
+  sudo sed -i "s,SONOBUOY_E2E_REGISTRY,${SONOBUOY_E2E_REGISTRY},g" /etc/eks/sonobuoy-e2e-registry-config
 fi
 
 ################################################################################
 ### ECR CREDENTIAL PROVIDER ####################################################
 ################################################################################
-if vercmp "$KUBERNETES_VERSION" gteq "1.22.0"; then
+if vercmp "${KUBERNETES_VERSION}" gteq "1.22.0"; then
   ECR_BINARY="ecr-credential-provider"
-  if [[ -n "$AWS_ACCESS_KEY_ID" ]]; then
+  if [[ -n "${AWS_ACCESS_KEY_ID}" ]]; then
     echo "AWS cli present - using it to copy ecr-credential-provider binaries from s3."
-    aws s3 cp --region $BINARY_BUCKET_REGION $S3_PATH/$ECR_BINARY .
+    aws s3 cp --region "${BINARY_BUCKET_REGION}" "${S3_PATH}/${ECR_BINARY}" .
   else
     echo "AWS cli missing - using wget to fetch ecr-credential-provider binaries from s3. Note: This won't work for private bucket."
-    sudo wget "$S3_URL_BASE/$ECR_BINARY"
+    sudo wget "${S3_URL_BASE}/${ECR_BINARY}"
   fi
-  sudo chmod +x $ECR_BINARY
+  sudo chmod +x "${ECR_BINARY}"
   sudo mkdir -p /etc/eks/ecr-credential-provider
-  sudo mv $ECR_BINARY /etc/eks/ecr-credential-provider
+  sudo mv "${ECR_BINARY}" /etc/eks/ecr-credential-provider
 
   # copying credential provider config file to eks folder
-  sudo mv $TEMPLATE_DIR/ecr-credential-provider-config /etc/eks/ecr-credential-provider/ecr-credential-provider-config
+  sudo mv "${TEMPLATE_DIR}/ecr-credential-provider-config" /etc/eks/ecr-credential-provider/ecr-credential-provider-config
 fi
 
 ################################################################################
 ### Cache Images ###############################################################
 ################################################################################
-if [[ "$CACHE_CONTAINER_IMAGES" == "true" && "$BINARY_BUCKET_REGION" != "us-iso-east-1" && "$BINARY_BUCKET_REGION" != "us-isob-east-1" ]]; then
+if [[ "${CACHE_CONTAINER_IMAGES}" == "true" && "${BINARY_BUCKET_REGION}" != "us-iso-east-1" && "${BINARY_BUCKET_REGION}" != "us-isob-east-1" ]]; then
   AWS_DOMAIN=$(imds 'latest/meta-data/services/domain')
   ECR_URI=$(/etc/eks/get-ecr-uri.sh "${BINARY_BUCKET_REGION}" "${AWS_DOMAIN}")
 
   PAUSE_CONTAINER="${ECR_URI}/eks/pause:${PAUSE_CONTAINER_VERSION}"
-  cat /etc/eks/containerd/containerd-config.toml | sed s,SANDBOX_IMAGE,$PAUSE_CONTAINER,g | sudo tee /etc/eks/containerd/containerd-cached-pause-config.toml
+  sed "s,SANDBOX_IMAGE,${PAUSE_CONTAINER},g" /etc/eks/containerd/containerd-config.toml | sudo tee /etc/eks/containerd/containerd-cached-pause-config.toml
```

Comment:
It is not about in-place replacement, so we could replace `cat FILE | sed ...` with `sed ... FILE`.